nft-rfc-006.md

```diff
@@ -0,0 +1,860 @@
+---
+nft-rfc: 6
+title: NFT Identifiers
+stage: draft
+category: NFT/METADATA
+author: Joe Andrieu @jandrieu
+created: 2021-01-21
+modified: 
+---
+
+# NFT RFC 006 NFT Identifiers (IIDs)
+
+Identifiers are used to refer to both on-chain and off-chain resources.
+Non-fungible tokens (NFTs) require a cryptographically secure means to
+both refer to specific tokens and prove control or ownership without
+reliance on a trusted third party. This RFC outlines the requirements
+for IIDs, proposes a new identifier approach, and compares this approach
+to existing and legacy identifier efforts.
+
+We refer to this new class of identifiers as Interchain Identifiers, or
```

Comment:
Should we be defining IID's this early in the project's life cycle? 

Is it necessary to create a new spec based on DID without seeing how we can extend the DID spec?

Can we satisfy our requirements with DIDs?

It could end up making IIDs not interoperable with other DIDs without some translation tool, which would add to the overhead of defining a new standard :thinking: 

https://w3c.github.io/did-spec-registries/#did-methods

Reply:
@PaddyMc I completely agree with your concern. I agree we don't want a new, incompatible identifier.

IIDs *are* DIDs. Or rather, a family of DIDs using DID Methods that share a few common properties that other DID Methods may not have. In some ways, this is similar to the Sidetree family of DID Methods: https://github.com/decentralized-identity/sidetree which currently includes both **did:ion** https://github.com/decentralized-identity/ion and **did:elem** https://github.com/decentralized-identity/element

In general, DID Methods are only expected to be interoperable syntactically. Each Method is free to innovate, both on how to access a given registry, and what properties are included in that Method's DID Document.  **did:ion**, for example, adds a "type" property. For interoperability, such properties should be added to the DID Spec Registries (which you link to). For example, there are a healthy family of Verification relationships, Verification method properties, and Verification method types already added. When a Method supports these properties, they SHOULD use the same definition as in the registry. 

If it isn't already clear, almost all methods are essentially NOT interoperable on the CRUD protocols for managing the DID Document. This is fundamental to how DIDs support Methods based on different registries or ledgers (some Methods don't even have a ledger).

This is described in more detail in the section on Extensibility https://w3c.github.io/did-core/#extensibility